### ISSUE TITLE
Metadata that use `year` as qualifier could cause issue with indexing

### ISSUE THREAD

**Issue by abollini** [DSpace/DSpace#9694]
## Describe the bug
If you define a custom metadata that uses year as qualifier and fill this metadata with a value that is not a integer, let's say a full date or something like AY2023/24 etc. the indexing of the item fails.
This is due to the configuration of a dynamic field in solr *.year as sint that will conflict with the field used to store the metadata for projection, see https://github.com/DSpace/DSpace/blob/main/dspace/solr/search/conf/schema.xml#L327

## To Reproduce
1. create a metadata local.test.year
2. add this metadata with the value AY2023/24
3. the item will disappear from solr and it is not longer searchable

## Expected behavior
The metadata should not impact the search or the administrator should be informed that year is a reserved word and the system should prevent the creation of this metadata. The last approach, prevent the creation of metadata with year as qualifier, will be probably the easier to implement

**Comment by tdonohue** [comment]
Agreed, this sounds like an annoying issue, as it's unexpected that using a qualifier named "year" would cause indexing issues.

Another option to consider:
* Is that dynamic Solr field `*.year` is even necessary in DSpace? If it's not serving a purpose, it might be removed.  If it is serving a purpose, then we might consider updating the wildcard so that it no longer "conflicts" with metadata fields that have a `year` qualifier.

Moving this to our 8.x/7.x maintenance board.  Needs a volunteer.
